engines/errors.go

```diff
@@ -26,6 +26,9 @@ var ErrImmutableMountNotSupported = errors.New("The engine doesn't support immut
 // doesn't exist.
 var ErrResourceNotFound = errors.New("The referenced resource wasn't found")
 
+// ErrHandlerInterrupt is returned when a handler that was given returns and error
+var ErrHandlerInterrupt = errors.New("Handler returned an error and interrupted iteration")
```

Comment:
This error (like ErrResourceNotFound) seems not to contain relevant context. However, if we log the reference used in the log message (and store it outside the error) then that would be ok. It just feels wrong to have an error type that doesn't store the context it relates to, as it is not inspectable programmatically. However, if the context is logged at least, that is better than nothing.